data/traccc_data_get_files.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+#
+# (c) 2023 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+#
+# Script downloading the traccc data file(s) through HTTPS, and unpacking them.
+#
+
+# Stop on errors.
+set -e
+set -o pipefail
+
+# Function printing the usage information for the script.
+usage() {
+   echo "Script downloading/unpacking data TGZ/MD5 files"
+   echo ""
+   echo "Usage: traccc_data_get_files.sh [options]"
+   echo ""
+   echo "Options:"
+   echo "  -f <filename>        Name of the data file, without its extension"
+   echo "  -d <webDirectory>    Directory holding the data and MD5 files"
+   echo "  -o <dataDirectory>   Main data directory"
+   echo "  -c <cmakeExecutable> CMake executable to use in the script"
+   echo "  -w <wgetExecutable> WGet executable to use in the script"
+   echo ""
+}
+
+# Default script arguments.
+TRACCC_DATA_NAME=${TRACCC_DATA_NAME:-"traccc-data-v1"}
+TRACCC_WEB_DIRECTORY=${TRACCC_WEB_DIRECTORY:-"https://acts.web.cern.ch/traccc/data"}
+TRACCC_DATA_DIRECTORY=${TRACCC_DATA_DIRECTORY:-$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)}
+TRACCC_CMAKE_EXECUTABLE=${TRACCC_CMAKE_EXECUTABLE:-cmake}
+TRACCC_WGET_EXECUTABLE=${TRACCC_WGET_EXECUTABLE:-wget}
+
+# Parse the command line argument(s).
+while getopts ":f:d:o:c:wh" opt; do
+   case $opt in
+      f)
+         TRACCC_DATA_NAME=$OPTARG
+         ;;
+      d)
+         TRACCC_WEB_DIRECTORY=$OPTARG
+         ;;
+      o)
+         TRACCC_DATA_DIRECTORY=$OPTARG
+         ;;
+      c)
+         TRACCC_CMAKE_EXECUTABLE=$OPTARG
+         ;;
+      w)
+         TRACCC_WGET_EXECUTABLE=$OPTARG
+         ;;
+      h)
+         usage
+         exit 0
+         ;;
+      :)
+         echo "Argument -$OPTARG requires a parameter!"
+         usage
+         exit 1
+         ;;
+      ?)
+         echo "Unknown argument: -$OPTARG"
+         usage
+         exit 1
+         ;;
+   esac
+done
+
+# Go into the target directory.
+cd "${TRACCC_DATA_DIRECTORY}"
+
+# Download the TGZ and MD5 files.
+"${TRACCC_WGET_EXECUTABLE}" --tries=5 --retry-connrefused --waitretry=10 \
+   "${TRACCC_WEB_DIRECTORY}/${TRACCC_DATA_NAME}.tar.gz"
+"${TRACCC_WGET_EXECUTABLE}" --tries=5 --retry-connrefused --waitretry=10 \
+   "${TRACCC_WEB_DIRECTORY}/${TRACCC_DATA_NAME}.md5"
```

Comment:
Could we split these files up so it becomes possible to download only a part of the testing data? That would reduce the load on wherever we host this, and make it easier as well as quicker to partially download the data (say, in the CI).

Reply:
:thinking: I'm not a big fan of what that would mean for versioning these files. Yes, the one TGZ file is pretty big. But at least then you only have a single file, with a single version number.

Reply:
Can't we put the files in a versioned directory, i.e. `https://acts.web.cern.ch/traccc/data/v4/tml_full.tar.gz`?

Reply:
We could. But again, I don't like the added complication. :frowning:

We could make TGZ files just meant for the CI. :thinking: So that CI jobs only download the files that they really need. While the download script by default downloads the "big file".

Reply:
I had a quick look, using commands like:

```sh
git grep "tml_pixel_barrel" -- "tests/*"
```

The **only** top level directory that we don't pick up anything from in our tests is `tml_pixels/`. From everything else we pick up at least some stuff.

At this stage it could still be a good idea to make a "CI specific" tarball with just the files that are actually used. This will need some detective work, but it could make the "CI specific" tarball a fraction of the size of the full one. But it's clear that simply splitting the tarballs by the top level directories would win us nothing with the CI.